juju/juju
# Add spaces reload functionality to network domain

This patch adds the reload spaces functionality (which still exists on the environs) to the new dqlite network domain. This was only possible thanks to the work introduced by the provider spaces in https://github.com/juju/juju/pull/17165.

The only changes outside the scope of the network domain, are the logger interfaces on the service factory, which has been updated with the Info and Trace levels.


## Checklist



## QA steps

The `ReloadSpaces()` method will be wired-up in a following PR at the bootstrap worker, so for the moment we only have to ensure that the unit tests pass:
```
TEST_PACKAGES="./domain/network/... -count=1 -race -check.vv" make run-go-tests
```


## Links



**Jira card:** JUJU-5831
